apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -104,35 +109,37 @@ export default class GenerateAction extends BaseRushAction {
       fsx.unlinkSync(tempShrinkwrapFilename);
     }
 
-    if (isLazy) {
-      console.log(colors.green(
-        `${os.EOL}Rush is running in "--lazy" mode. ` +
-        `You will need to run a normal "rush generate" before committing.`));
+    if (isLazy || this.rushConfiguration.packageManager === 'pnpm') {
```

Comment:
>this.rushConfiguration.packageManager === 'pnpm' [](start = 18, length = 48)

Your logic assumes the only possibilities are NPM or PNPM.  I find it a little confusing to think about two different ways this of testing this. 


Maybe you could just test PNPM everywhere?